pkg/controllers/dashboard/systemcharts/controller_test.go

```diff
@@ -168,42 +209,283 @@ func Test_ChartInstallation(t *testing.T) {
-				t.Errorf("handler.onRepo() error = %v, wantErr %v", err, tt.wantErr)
+				require.FailNow(t, "handler.onRepo() error = %v, wantErr %v", err, tt.wantErr)
 				return
 			}
 		})
 	}
 }
+
+func Test_relatedConfigMaps(t *testing.T) {
+	const fooMap = "foo"
+	orig := settings.ConfigMapName.Get()
+	defer func() { settings.ConfigMapName.Set(orig) }()
+	settings.ConfigMapName.Set(fooMap)
+	tests := []struct {
+		changedObj runtime.Object
+		name       string
+		want       []relatedresource.Key
+	}{
+		{
+			name: "rancher-config change",
+			changedObj: &v1.ConfigMap{ObjectMeta: metav1.ObjectMeta{
+				Name:      chart.CustomValueMapName,
+				Namespace: namespace.System,
+			}},
+			want: []relatedresource.Key{{Name: repoName, Namespace: ""}},
+		},
+		{
+			name: "configMap from settings change",
+			changedObj: &v1.ConfigMap{ObjectMeta: metav1.ObjectMeta{
+				Name:      fooMap,
+				Namespace: namespace.System,
+			}},
+			want: []relatedresource.Key{{Name: repoName, Namespace: ""}},
+		},
+		{
+			name: "rancher-config changed wrong namespace",
+			changedObj: &v1.ConfigMap{ObjectMeta: metav1.ObjectMeta{
+				Name:      chart.CustomValueMapName,
+				Namespace: "",
+			}},
+			want: nil,
+		},
+		{
+			name: "configMap from settings change wrong namespace",
+			changedObj: &v1.ConfigMap{ObjectMeta: metav1.ObjectMeta{
+				Name:      fooMap,
+				Namespace: fooMap,
+			}},
+			want: nil,
+		},
+		{
+			name: "incorrect type",
+			changedObj: &v3.Setting{ObjectMeta: metav1.ObjectMeta{
+				Name:      chart.CustomValueMapName,
+				Namespace: namespace.System,
+			}},
+			want: nil,
+		},
+	}
+	for i := range tests {
+		test := tests[i]
+		t.Run(test.name, func(t *testing.T) {
+			got, err := relatedConfigMaps("", "", test.changedObj)
+			require.NoError(t, err, "unexpected error")
+			require.Equal(t, test.want, got)
+
+		})
+	}
+}
+
+func Test_relatedFeature(t *testing.T) {
+	tests := []struct {
+		changedObj runtime.Object
+		name       string
+		want       []relatedresource.Key
+	}{
+		{
+			name: "feature in map",
+			changedObj: &v3.Feature{ObjectMeta: metav1.ObjectMeta{
+				Name: features.MCM.Name(),
+			}},
+			want: []relatedresource.Key{{Name: repoName, Namespace: ""}},
+		},
+		{
+			name: "feature not in map",
+			changedObj: &v3.Feature{ObjectMeta: metav1.ObjectMeta{
+				Name: "foo",
+			}},
+			want: nil,
+		},
+		{
+			name: "incorrect type",
+			changedObj: &v3.Setting{ObjectMeta: metav1.ObjectMeta{
+				Name:      chart.CustomValueMapName,
+				Namespace: namespace.System,
+			}},
+			want: nil,
+		},
+	}
+	for i := range tests {
+		test := tests[i]
+		t.Run(test.name, func(t *testing.T) {
+			got, err := relatedFeatures("", "", test.changedObj)
+			require.NoError(t, err, "unexpected error")
+			require.Equal(t, test.want, got)
+
+		})
+	}
+}
+
+func Test_relatedSettings(t *testing.T) {
+	tests := []struct {
+		changedObj runtime.Object
+		name       string
+		want       []relatedresource.Key
+	}{
+		{
```

Comment:
Nit: I'd recommend that we have a separate case for each setting in the map, rather than only one case for the item in the map.